scimusmn/app-template
# Extch 348 gatsby upgrade to v5

The P.R. is the result of migrating app-template from Gatsby v4 to v5, and accounting for all the other necessary upgrades. 

Important notes:
1. You need to be using Node v18
2. The new GraphQL engine has slightly changed the way they handle sort and aggregation fields (see example here: https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v4-to-v5/#graphql-schema-changes-to-sort-and-aggregation-fields)
3. The built-in GraphiQL tool has a new cool interface 

To test
- Pull down this branch
- Set node version to 18
- Run `yarn` to install packages
- Run `yarn develop` and confirm base application is working
- Run `yarn install:contentful` and follow prompts to connect with existing Contentful Space/Content Type
- Run `yarn develop` and check that pages are being rendered based on the content type you selected
- Run `yarn install:arduino-base` to install the arduino-base submodel
- Run `yarn develop` and check that the [test page](https://d.pr/i/PLfKRq) is rendered (http://localhost:3000/arduino/)
- (Optional) If you have an arduino, connect and relaunch app to test communication


### Preparation checklist

* [x] Merge `main` into your branch and resolve resulting conflicts
* [x] Select one or multiple reviewers who are well suited to discuss your changes
* [ ] Write a new programmatic test, if warranted
* [ ] Edit or create the documentation necessary for others to make use of your changes

### Writing a helpful description

* [x] Link to the specific user-story being addressed, and describe how you've addressed it
* [x] Describe how your changes should be tested
* [ ] Link to specific files, line-numbers, and decisions you would like feedback on
* [ ] Bonus: Include screenshots or screen-captures to describe any visual changes
